dashboard/app/models/user.rb

```diff
@@ -494,6 +494,7 @@ def user_level_for(script_level, level)
 
   def user_level_locked?(script_level, level)
     return false unless script_level.stage.lockable?
+    return false if authorized_teacher?
```

Comment:
It could. I'm not sure one approach is better than the author. Either way `user_level_locked?` needs to check for `authorized_teacher?`. It seems a little silly to fall through to user_level in the case where we already know the answer.